pkg/route/validation/validation_test.go

```diff
@@ -843,6 +981,522 @@ func TestValidateRoute(t *testing.T) {
+			headerType:           "response",
+		},
+		{
+			name: "Valid response headers",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcardpolicy",
+					Namespace: "foo",
+				},
+				Spec: routev1.RouteSpec{
+					Host:           "subdomain.wildcard.test",
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
+					HTTPHeaders: &routev1.RouteHTTPHeaders{
+
+						Actions: routev1.RouteHTTPHeaderActions{
+							Response: []routev1.RouteHTTPHeader{
+								{
+									Name: headerNameXFrame,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "DENY",
+										},
+									},
+								},
+								{
+									Name: headerNameXSS,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "1;mode=block",
+										},
+									},
+								},
+								{
+
+									Name: headerNameXSS,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Delete,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrorMessage: "spec.httpHeaders.actions.response[2].name: Invalid value: \"X-XSS-Protection\": header name has been repeated",
+			headerType:           "response",
+		},
+		{
+			name: "Valid request headers",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcardpolicy",
+					Namespace: "foo",
+				},
+				Spec: routev1.RouteSpec{
+					Host:           "subdomain.wildcard.test",
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
+					HTTPHeaders: &routev1.RouteHTTPHeaders{
+
+						Actions: routev1.RouteHTTPHeaderActions{
+							Request: []routev1.RouteHTTPHeader{
+								{
+									Name: "Accept",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "text/plain,text/html",
+										},
+									},
+								},
+								{
+
+									Name: "Accept-Encoding",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Delete,
+									},
+								},
+								{
+									Name: "Condtional",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "%[req.hdr(Host)] if foo",
```

Comment:
The EP says that conditional is not allowed.  This should be an error.

Reply:
It is not allowed however in the future we might allow it. For now we are focusing allowing it as a string which will not functional and be interpreted by HAProxy and also it should not crash the router for which we do escape and quoting.

Reply:
This was already discussed on slack.